### PR TITLE
React18v4

### PR DESCRIPTION
**Pull request by bunsenstraat**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -166,6 +166,7 @@ jobs:
           at: .
       - run: unzip ./persist/dist.zip
       - run: yarn install --cwd ./apps/remix-ide-e2e --modules-folder ../../node_modules
+      - run: mkdir node_modules/hardhat && wget https://unpkg.com/hardhat/console.sol -O node_modules/hardhat/console.sol
```
Comment (yann300) [inline .circleci/config.yml:169]: why do you need to do this?

File: apps/remix-ide-e2e/src/helpers/init.ts [hunk 2]
```diff
@@ -27,30 +27,36 @@ export default function (browser: NightwatchBrowser, callback: VoidFunction, url
...
               display:none !important;
             }
           `);
         })
-      }
+    }})
+    .perform(() => {
+      browser.execute(function () { 
+        (window as any).logs = []
+        console.log = function () {
```
Comment (yann300) [inline apps/remix-ide-e2e/src/helpers/init.ts:51]: is this a new feature for the CI?

File: libs/remix-ui/run-tab/src/lib/actions/actions.ts [hunk 3]
```diff
@@ -88,6 +88,7 @@ export const updateScenarioPath = (dispatch: React.Dispatch<any>, path: string)
 }
 
 export const setSendTransactionValue = (dispatch: React.Dispatch<any>, value: string) => {
+  console.log('setSendTransactionValue', value)
```
Comment (yann300) [inline libs/remix-ui/run-tab/src/lib/actions/actions.ts:91]: to remove?

File: .circleci/config.yml (hunk 1 above)
Comment (bunsenstraat) [reply .circleci/config.yml:169]: the console.sol is slow to load from unpkg in e2e. it caused issues

Comment (bunsenstraat) [reply .circleci/config.yml:169]: this puts it where the localhost can find is using the nx_npm_url param

File: apps/remix-ide-e2e/src/helpers/init.ts (hunk 2 above)
Comment (bunsenstraat) [reply apps/remix-ide-e2e/src/helpers/init.ts:51]: yeah you can fetch the logs then in any test. helpful but maybe not necessary for this PR.

**Review by yann300 — APPROVED** [review]

**Comment by drafish** [comment]
https://github.com/ethereum/remix-project/blob/react18v4/package.json#L369-L371

```
  "resolutions": {
    "@types/react": "^17.0.24"
  }
```

This should be removed. I added this to fix typescript error in #2581 . Because it will default to be ^18 . Now @bunsenstraat update react version to ^18, this should be removed.

**Pull request closed without merging**